lib/src/util/withClass.tsx

```diff
@@ -4,16 +4,27 @@
  * the terms of the Eclipse Public License 2.0 which accompanies this
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
-import React, { FunctionComponent } from 'react';
+import React, { forwardRef, DetailedHTMLProps, SVGProps } from 'react';
 import classnames from 'classnames';
 
+type NativeElType<E extends keyof JSX.IntrinsicElements> =
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  JSX.IntrinsicElements[E] extends DetailedHTMLProps<any, infer A>
+    ? A
+    : JSX.IntrinsicElements[E] extends SVGProps<infer B>
+      ? B
+      : never;
```

Comment:
I prefer that multi-line ternaries be formatted like this for readability:

```suggestion
  JSX.IntrinsicElements[E] extends DetailedHTMLProps<any, infer A> ? A : 
  JSX.IntrinsicElements[E] extends SVGProps<infer B> ? B : 
  never;
```